### ISSUE TITLE
appendText in MarkdownString does not seem to escape greater than and less than symbols

### ISSUE THREAD

**Issue by rcjsuen** [microsoft/vscode#124874]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.56.2
- OS Version: macOS 10.15.7

Steps to Reproduce:

```TypeScript
import * as vscode from 'vscode';

export function activate(context: vscode.ExtensionContext) {

	const provider = vscode.languages.registerHoverProvider("plaintext", {
		provideHover() {
			const markdownString = new vscode.MarkdownString();
			markdownString.appendText("U+0008, <control>")
			return new vscode.Hover(markdownString);
		}
	});

	context.subscriptions.push(provider);
}
```

1. Create an extension with the hover provider above.
2. Create a plain text `test.txt` file and put some content in it.
3. Hover over it. Notice that the `<control>` string is missing.

![image](https://user-images.githubusercontent.com/15629116/120049756-f8a7da80-bfe8-11eb-910a-bab0d7a32928.png)

https://github.com/microsoft/vscode/blob/887ff91a14ad2c76cdc5a776fe10bfd7494fef6e/src/vs/vscode.d.ts#L2592-L2596

It seems like the `<` and `>` in `<control>` isn't getting escaped properly...?

**Comment by rcjsuen** [comment]
```TypeScript
markdownString.appendText("U+0008, &lt;control&gt;")
```
If I use this instead it works.

![image](https://user-images.githubusercontent.com/15629116/120050474-6fde6e00-bfeb-11eb-9b5d-b63a5b2cff55.png)

**Comment by jrieken** [comment]
Some time passed but his works for me... Can you try again with [VS Code Insiders](https://code.visualstudio.com/insiders/)?

<img width="284" alt="Screen Shot 2021-09-27 at 16 13 59" src="https://user-images.githubusercontent.com/1794099/134925934-2cd1093d-0d5d-48e9-8806-170436c2c6d5.png">

**Comment by rcjsuen** [comment]
Version: 1.61.0-insider (Universal)
Commit: 5991e3c4b97eca5568069bf3e0910ab693ad3549
Date: 2021-09-27T05:14:53.754Z (10 hrs ago)
Electron: 13.4.0
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0

-------

@jrieken I have confirmed that it is working with the build above. Perhaps it was fixed by https://github.com/microsoft/vscode/pull/132182...?

**Comment by jrieken** [comment]
> Perhaps it was fixed by #132182...?

Unlikely but maybe...
